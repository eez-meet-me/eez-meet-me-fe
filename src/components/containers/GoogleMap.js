import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Map, GoogleApiWrapper } from 'google-maps-react';
import Markers from '../marker/Markers';
import InformationWindow from '../infoWindow/InfoWindow';
import { getPinsList } from '../../selectors/pinSelectors';
import { getPins } from '../../actions/pinActions';
import Header from '../header/Header';
import MarkerList from '../markerList/MarkerList';

class GoogleMap extends Component {

  static propTypes = {
    google: PropTypes.object.isRequired,
    fetch: PropTypes.func.isRequired,
    pins: PropTypes.array.isRequired
  }

  componentDidMount() {
    this.props.fetch();
    setInterval(() => {
      this.props.fetch();
    }, 10000);
  }

  shouldComponentUpdate(nextProps, nextState) {
    return nextState.activeMarker !== this.state.activeMarker ||
      nextProps.pins.length !== this.props.pins.length ||
      !nextProps.pins.every((pin, i) => pin._id === this.props.pins[i]._id);
  }

  state = {
    showingInfoWindow: false,
    activeMarker: {},
    selectedPlace: {}
  }

  onMarkerClick = (props, marker, e) => {
    this.setState({
      selectedPlace: props,
      activeMarker: marker,
      showingInfoWindow: true
    });
  }

  onClose = props => {
    if(this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: {}
      });
    }
  }

  render() {
    const { activeMarker, showingInfoWindow } = this.state;
    const { pins } = this.props;
    return (
      <>
      <Header />
        <Map
          google={this.props.google}
          zoom={8}
          style={{ width: '100%', height: '50%' }}
          initialCenter={{ lat: 45.5234166, lng: -122.6808846 }}
        >
          <Markers onClick={this.onMarkerClick} pins={pins} />
          <InformationWindow
            marker={activeMarker}
            visible={showingInfoWindow}
            onClose={this.onClose}
            pins={pins} />
        </Map>
        <MarkerList pins={pins} />
      </>
    );
  }
}

const mapStateToProps = state => ({
  pins: getPinsList(state)
});

const mapDispatchToProps = dispatch => ({
  fetch() {
    dispatch(getPins());
  }
});

const ConnectedGoogleMap = connect(
  mapStateToProps,
  mapDispatchToProps
)(GoogleMap);

export default GoogleApiWrapper({
  apiKey: process.env.API_KEY
})(ConnectedGoogleMap);
