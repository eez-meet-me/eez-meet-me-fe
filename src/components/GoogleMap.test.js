import React from 'react';
import { shallow } from 'enzyme';
import GoogleMap from './containers/GoogleMap';

describe('GoogleMap component', () => {
  it('renders GoogleMap', () => {
    const wrapper = shallow(<GoogleMap />);
    expect(wrapper).toMatchSnapshot();
  });
});
