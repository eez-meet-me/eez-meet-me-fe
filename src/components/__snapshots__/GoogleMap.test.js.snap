// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoogleMap component renders GoogleMap 1`] = `<DefaultLoadingContainer />`;
